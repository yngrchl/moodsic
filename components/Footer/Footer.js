import React from 'react';

/**
 * Footer at bottom of each page
 */
const Footer = () => (
  <footer>
    <p>source code at</p>{" "}
    <a href="https://github.com/yngrchl/moodsic">
      <img src="github-icon.png" alt="Github logo" />
    </a>
  </footer>
);

export default Footer;
