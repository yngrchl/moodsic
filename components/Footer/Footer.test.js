import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import Footer from './Footer';

describe('Footer', () => {
  it('renders Footer unchanged', () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });

  it('has a link to Github source code', () => {
    render(<Footer />);

    const githubLink = screen.getByRole('link');

    expect(githubLink).toHaveAttribute(
      'href',
      'https://github.com/yngrchl/moodsic',
    );
  });
});
