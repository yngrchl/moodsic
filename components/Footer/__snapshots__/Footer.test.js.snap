// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders Footer unchanged 1`] = `
<div>
  <footer>
    <p>
      source code at
    </p>
     
    <a
      href="https://github.com/yngrchl/moodsic"
    >
      <img
        alt="Github logo"
        src="github-icon.png"
      />
    </a>
  </footer>
</div>
`;
